acceptance/lib/pxp-agent/test_helper.rb

```diff
@@ -520,25 +519,34 @@ def check_puppet_non_blocking_response(identity, transaction_id, max_retries, qu
         # allow a few unknowns before giving up
         unknown_count += 1
       else
-        puppet_run_result = JSON.parse(action_result['stdout'])['status']
-        puppet_run_environment = JSON.parse(action_result['stdout'])['environment']
+        run_result = JSON.parse(action_result['stdout'])
```

Comment:
Oh, we do expect responses without the stdout key. Those are generally either error_message or a response that says the task is still running. We loop until we get a response with results and status of finished, or until we time out.